consul/data_source_consul_peering.go

```diff
@@ -0,0 +1,103 @@
+package consul
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceConsulPeering() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceConsulPeeringRead,
+		Schema: map[string]*schema.Schema{
+			// In
+			"peer_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"datacenter": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"partition": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			// Out
+			"deleted_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"meta": {
+				Type:     schema.TypeMap,
+				Computed: true,
+
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"peer_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"peer_ca_pems": {
+				Type:     schema.TypeList,
+				Computed: true,
+
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"peer_server_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"peer_server_addresses": {
+				Type:     schema.TypeList,
+				Computed: true,
+
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
```

Comment:
We also added ImportedServiceCount and ExportedServiceCount fields to the Peering type in the api package. Should those also be included in the data source?